huntsman/observatory.py

```diff
@@ -232,202 +238,83 @@ def autofocus_cameras(self, *args, **kwargs):
 
         return result
 
-    def take_evening_flats(self,
-                           alt=None,
-                           az=None,
-                           min_counts=5000,
-                           max_counts=15000,
-                           bias=1000,
-                           max_exptime=60.,
-                           camera_list=None,
-                           target_adu_percentage=0.5,
-                           max_num_exposures=10,
-                           *args, **kwargs
-                           ):
-        """Take flat fields
+    def take_flat_fields(self, camera_names=None, alt=None, az=None,
+                         safety_func=None, **kwargs):
+        """
+        Take flat fields for each camera in each filter, respecting filter order.
 
         Args:
-            alt (float, optional): Altitude for flats
-            az (float, optional): Azimuth for flats
-            min_counts (int, optional): Minimum ADU count
-            max_counts (int, optional): Maximum ADU count
-            bias (int, optional): Default bias for the cameras
-            max_exptime (float, optional): Maximum exposure time before stopping
-            camera_list (list, optional): List of cameras to use for flat-fielding
-            target_adu_percentage (float, optional): Exposure time will be adjust so
-                that counts are close to: target * (`min_counts` + `max_counts`). Default
-                to 0.5
-            max_num_exposures (int, optional): Maximum number of flats to take
-            *args (TYPE): Description
-            **kwargs (TYPE): Description
-
+            camera_names (list, optional): List of camera names to take flats with.
+                Default to `None`, which uses all cameras.
+            alt (float, optional): Altitude for flats in degrees. Default `None` will use the
+                `flat_field.twilight.alt` config value.
+            az (float, optional): Azimuth for flats in degrees. Default `None` will use the
+                `flat_field.twilight.az` config value.
+            min_counts (int, optional): Minimum ADU count. Default 5000.
+            max_counts (int, optional): Maximum ADU count. Default 15000.
+            bias (int, optional): Default bias in ADU counts for the cameras. Default 32.
+            target_scaling (float, optional): Required to be between [0,1] so
+                target_adu is proportionally between 0 and digital saturation level.
+                Default 0.17.
```

Comment:
@AnthonyHorton said something about filling the CCD to ~1/6th of its capacity for the flat fields. `bias=32` is close to what @JAAlvarado-Montes was getting from his ZWO biases, but this will be made to be camera-specific in the future.